platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/StellarPaymentObserver.java

```diff
@@ -86,6 +127,44 @@ private void restart() {
     }
   }
 
+  SSEStream<OperationResponse> startSSEStream() {
+    String latestCursor = fetchStreamingCursor();
+    PaymentsRequestBuilder paymentsRequest =
+        server
+            .payments()
+            .includeTransactions(true)
+            .cursor(latestCursor)
+            .order(RequestBuilder.Order.ASC)
+            .limit(MAX_RESULTS);
+
+    return paymentsRequest.stream(
+        new EventListener<>() {
+          @Override
+          public void onEvent(OperationResponse operationResponse) {
+            debugF("received event {}", operationResponse.getId());
+            lastActivityTime = Instant.now();
+            handleEvent(operationResponse);
+          }
+
+          @Override
+          public void onFailure(Optional<Throwable> exception, Optional<Integer> statusCode) {
+            handleFailure(exception, statusCode);
+          }
+        });
+  }
+
+  void handleFailure(Optional<Throwable> exception, Optional<Integer> statusCode) {
+    // The SSEStreamer has internal errors. We will give up and let the container manager to
+    // restart.
+    Log.errorEx("stellar payment observer error: ", exception.get());
+
+    // Stop the SSEStream
+    stream.close();
+
+    // Mark the observer unhealthy
+    healthy = false;
```

Comment:
We already have a function called `restart()` that pauses the streamer for some (exponential increasing) time and then restarts the streamer after that time has passed. Why don't we just call `restart()`? 

Marking the job as unhealthy in the first failure and not adding any way for the software to recover the job sounds pretty extreme.

Also, your solution relies on the assumption that k8s will detect the unhealthy state and restart the pod, which may not happen depending on the setup.

Also, restarting the pods would make the whole AP unavailable for 2 whole minutes, in case the observer and the sep-server are launched together and there's only a single pod. I feel like this solution will add new problems and would require a very specific deployment setup to work. It can cause more harm than good.